code/datums/components/gunpoint.dm

```diff
@@ -0,0 +1,103 @@
+#define GUNPOINT_SHOOTER_STRAY_RANGE 3
+#define GUNPOINT_DELAY_STAGE_2 20
+#define GUNPOINT_DELAY_STAGE_3 80 // cumulative with past stages, so 100 deciseconds
+#define GUNPOINT_MULT_STAGE_1 1
+#define GUNPOINT_MULT_STAGE_2 2
+#define GUNPOINT_MULT_STAGE_3 2.5
+
+
+/datum/component/gunpoint
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+
+	var/mob/living/shooter
+	var/mob/living/target
+	var/obj/item/gun/weapon
+
+	var/stage = 1
+	var/datum/status_effect/status_hold_up
+	var/datum/status_effect/status_held_up
+	var/damage_mult = GUNPOINT_MULT_STAGE_1
+
+/datum/component/gunpoint/Initialize(mob/living/targ, obj/item/gun/wep)
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	shooter = parent
+	target = targ
+	weapon = wep
+	RegisterSignal(targ, list(COMSIG_MOB_ATTACK_HAND, COMSIG_MOB_ITEM_ATTACK, COMSIG_MOB_ATTACK_RANGED, COMSIG_MOVABLE_MOVED), .proc/nap_violation)
+	RegisterSignal(weapon, list(COMSIG_ITEM_DROPPED, COMSIG_ITEM_EQUIPPED), .proc/cancel)
+
+	shooter.visible_message("<span class='danger'>[shooter] aims [weapon] point blank at [target]!</span>", \
+		"<span class='danger'>You aim [weapon] point blank at [target]!</span>", target)
+	to_chat(target, "<span class='userdanger'>[shooter] aims [weapon] point blank at you!</span>")
+	target.emote("gasp") // *extremely bad russian accent* no!
```

Comment:
this is dumb, it takes away the ability for players to reply to the action themselves because you force them to gasp, and this'll cause shit like Ian to gasp because you're holding them at gunpoint.

Reply:
1. no??? 2. feature

add a MOB_ORGANIC check